src/WebJobs.Script.WebHost/Metrics/FlexConsumptionMetricsPublisher.cs

```diff
@@ -24,6 +20,7 @@ public class FlexConsumptionMetricsPublisher : IMetricsPublisher, IDisposable
         private readonly ILogger<FlexConsumptionMetricsPublisher> _logger;
         private readonly object _lock = new object();
         private readonly IFileSystem _fileSystem;
+        private readonly LegionMetricsFileManager _metricsFileManager;
```

Comment:
Both Flex and CV1 on Legion use the same file writer for metrics, so I refactored this logic out